cluster/addons.go

```diff
@@ -38,15 +44,113 @@ func (c *Cluster) deployK8sAddOns(ctx context.Context) error {
 func (c *Cluster) deployUserAddOns(ctx context.Context) error {
 	log.Infof(ctx, "[addons] Setting up user addons..")
 	if c.Addons == "" {
-		log.Infof(ctx, "[addons] No user addons configured..")
-		return nil
+		return c.deployAddonsInclude(ctx)
 	}
 
 	if err := c.doAddonDeploy(ctx, c.Addons, UserAddonResourceName); err != nil {
 		return err
 	}
 	log.Infof(ctx, "[addons] User addon deployed successfully..")
+	return c.deployAddonsInclude(ctx)
+}
+
+// deployAddonsInclude takes any passed addons from the addons key and appends yaml contents from urls/paths
+// then deploys all addons at once. Not doing this in this manner will fail to deploy all of the plugins.
+// Only the addons from the addons: key would be deployed.
+func (c *Cluster) deployAddonsInclude(ctx context.Context) error {
+	var userAddons = ""
+	log.Infof(ctx, "[addons] Checking for included user addons")
+
+	yamlDelimiter := regexp.MustCompile("(\n---$|\n---\n$)")
```

Comment:
is it possible to skip the regex check and check using the yaml library in golang

Reply:
May I ask what the reasoning for this is? I don't understand the approach you are wanting to take, and I don't see anything in the go yaml lib to do something like this.